crowbar_framework/app/models/backup.rb

```diff
@@ -0,0 +1,192 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "find"
+
+class Backup
+  include ActiveModel::Model
+
+  attr_accessor :name, :created_at, :filename, :path
+
+  validates_presence_of :name, :created_at
+  validates :created_at, exclusion: { in: [nil, ""] }
+  validate :filename, :filename_has_format
+  validate :filename, :filename_has_characters
+
+  def initialize(options)
+    @name = options.fetch :name, nil
+    @created_at = options.fetch :created_at, Time.zone.now.strftime("%Y%m%d-%H%M%S")
+    @filename = "#{@name}-#{@created_at}.tar.gz"
+    @path = Backup.image_dir.join(@filename)
+  end
+
+  def as_json(options: {})
+    result = super
+    result["path"] = path.to_s
+    result
+  end
+
+  def save
+    valid? && persist!
+  end
+
+  def exist?
+    !Backup.where(name: name, created_at: created_at).nil?
```

Comment:
How does that work? You already got an instance, so how should that ever be `nil`?